### PR TITLE
Add registry and connector resource support for synapse unit testing framework

### PR DESCRIPTION
**Pull request by sajithaliyanage**
## Purpose
From this PR it enables the access of registry resources and connector resources for the unit testing framework. To achieve this task, the registry has mocked in memory inside the synapse engine when the testing mode enabled and connectors will deploy temporary during the unit testing workflow.  Added an artifact undeployer function to the testing framework to undeploy all the artifacts deploying in the testing phase. 

## Goals
- Enable access to the registry resources for unit testing framework
- Enable access to the connector resources for the unit testing framework
- Artifact undeployer to undeploy configuration used for unit testing
- Improve unit testing response message

### CODE REVIEW

**Comment by sajithaliyanage** [comment]
Related issue : https://github.com/wso2/micro-integrator/issues/614

**Comment by sajithaliyanage** [comment]
Related issue - https://github.com/wso2/product-ei/issues/4408

File: modules/core/src/main/java/org/apache/synapse/unittest/CommonUtils.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+
```
Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/unittest/CommonUtils.java]: Formatting issue in license header
let's correct rest as well in this PR

Comment (sajithaliyanage) [reply modules/core/src/main/java/org/apache/synapse/unittest/CommonUtils.java]: License header issue fixed in https://github.com/wso2/wso2-synapse/pull/1452

**Comment by sajithaliyanage** [comment]
License header issue fixed in https://github.com/wso2/wso2-synapse/pull/1452

**Pull request merged**